doc/src/conf.py

```diff
@@ -0,0 +1,161 @@
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2018 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import os
+
+
+# -- General configuration ------------------------------------------------
+
+# minimal Sphinx version required.
+needs_sphinx = '1.2'
+
+# Sphinx extension module names.
+sys.path.append(os.path.abspath('custom'))  # path to custom extensions.
+extensions = [
+    'sphinx.ext.autodoc',
+    'sphinx.ext.doctest',
+    'sphinx.ext.intersphinx',
+    'sphinx.ext.todo',
+    'numfig',
+    'cylc_lang',
+]
+
+# Add any paths that contain templates.
+templates_path = ['_templates']
+
+# The suffix of source filenames.
+source_suffix = '.rst'
+
+# The master toctree document.
+master_doc = 'index'
+
+# General information about the project.
+project = u'The Cylc Suite Engine'
+copyright = u'2008-2018 NIWA & British Crown (Met Office) & Contributors'
+version = '7.8'  # The short X.Y version.
+release = '7.8.0'  # The full version, including alpha/beta/rc tags.
+
+# List of patterns, relative to source directory, that match files and
+# directories to ignore when looking for source files.
+exclude_patterns = ['_build']
+
+# The name of the Pygments (syntax highlighting) style to use.
+pygments_style = 'manni'
+
+# Enable automatic numbering of any captioned figures, tables & code blocks.
+# 'numfig' THIS REQUIRES v1.3+, & assume only v1.2 for now, so can't use.
+#numfig = True
+#numfig_secnum_depth = 0
+# Therefore must use the 'numfig' extension, configuring as such:
+figure_caption_prefix = "Fig. "
+
+
+# -- Options for HTML output ----------------------------------------------
+
+# The builtin HTML theme to build upon, with customisations to it. Notably
+# customise with a white 'sticky' sidebar; make headings & links text the Cylc
+# logo colours & make code block background the logo green made much lighter.
+# html_theme = "classic" -> not required to set, as this is the default.
```

Comment:
This comment does not speak the truth on my system. `html_theme = "classic"` needs to be uncommented, or i get errors like this:
```
Theme error:                                                         
unsupported theme option 'headbgcolor' given   
```

Reply:
(To provide context) it seems that for early versions of Sphinx, e.g. ``1.2`` as I developed this PR with, ``"classic"`` was taken as the default theme, & in fact *could not be* specified explicitly (I tried but it produced an error). But for later versions as you have observed it *needs* to be specified in the ``conf.py``. I noted the same when building on ``1.7`` on my VM from home.

Since we know we need to use a fairly up-to-date version (as outlined in https://github.com/cylc/cylc/pull/2910#issuecomment-452262395) I will indeed uncomment it as I update the PR to build cleanly for the newer versions.